include/fiction/algorithms/physical_design/on_the_fly_circuit_design_on_defective_surface.hpp

```diff
@@ -0,0 +1,216 @@
+        const auto miter = mockturtle::miter<technology_network>(mapped_network, *gate_level_layout);
+        const auto eq    = mockturtle::equivalence_checking(*miter);
+
+        // empty defect-surface is returned when equivalence check is empty
+        if (!eq.has_value())
+        {
+            return sidb_defect_surface<CellLyt>{};
+        }
+
+        // in case of equality, an sidb_defect_surface with the SiDBs of the circuit is returned
+        if (*eq)
+        {
+            sidb_defect_surface<CellLyt> sidbs_and_defects{cell_level_layout};
+
+            // add defects to the circuit.
+            params.parameterized_gate_library_parameter.defect_surface.foreach_sidb_defect(
+                [&sidbs_and_defects](const auto& defect)
+                { sidbs_and_defects.assign_sidb_defect(defect.first, defect.second); });
+
+            return sidbs_and_defects;
+        }
+
+        // in case of inequality, empty defect-surface is returned
+        return sidb_defect_surface<CellLyt>{};
+    }
+
+  private:
+    /**
+     * Gate-level layout.
+     */
+    GateLyt lattice_tiling;
+    /**
+     * Mapped network.
+     */
+    Ntk mapped_network{};
+    /**
+     * Parameters for the on-the-fly circuit design.
+     */
+    on_the_fly_circuit_design_params<CellLyt> params{};
+    /**
+     * Statistics for the on-the-fly circuit design.
+     */
+    on_the_fly_circuit_design_stats& stats;
+};
+
+}  // namespace detail
+
+/**
+ * This function applies the on-the-fly circuit design algorithm to a specified defective surface, automatically
+ * creating customized SiDB gates while integrating atomic defects as essential components of the design. It was
+ * proposed in \"On-the-fly Defect-Aware Design of Circuits based on Silicon Dangling Bond Logic\" by J. Drewniok, M.
+ * Walter, S. S. H. Ng, K. Walus, and R. Wille in IEEE NANO 2024.
+ *
+ * @tparam Ntk      The type of the input network.
+ * @tparam CellLyt  The type of the cell layout.
+ * @tparam GateLyt  The type of the gate layout.
+ * @param mapped_network  The input network to be mapped onto the defective surface.
+ * @param design_params   The parameters used for designing the circuit, encapsulated in an
+ * `on_the_fly_circuit_design_params` object.
+ * @param lattice_tiling  The lattice tiling used for the circuit design.
+ * @param defect_surface  The defective surface on which the circuit is to be designed.
+ * @param stats           Pointer to a structure for collecting statistics. If nullptr, statistics are not collected.
+ *
+ * @return A `sidb_defect_surface<CellLyt>` representing the designed circuit on the defective surface.
+ */
+template <typename Ntk, typename CellLyt, typename GateLyt>
+[[nodiscard]] sidb_defect_surface<CellLyt> on_the_fly_circuit_design_on_defective_surface(
+    const Ntk& mapped_network, const on_the_fly_circuit_design_params<CellLyt>& design_params,
+    const GateLyt& lattice_tiling, on_the_fly_circuit_design_stats* stats = nullptr)
```

Comment:
```suggestion
template <typename Ntk, typename CellLyt, typename GateLyt>
[[nodiscard]] sidb_defect_surface<CellLyt> on_the_fly_circuit_design_on_defective_surface(
    const Ntk& ntk, const on_the_fly_circuit_design_params<CellLyt>& params,
    const GateLyt& lattice_tiling, on_the_fly_circuit_design_stats* stats = nullptr)
```

Reply:
Swap `params` and `lattice_tiling` such that you can default-initialize `params = {}`.

Reply:
You don't specify anywhere that the network has to be mapped already. (and it should not have to be)